### PR TITLE
[Windows] Fix command line arguments limit using gcc

### PR DESCRIPTION
**Pull request by Tormund**
Use `gcc @file` feature at linker phase

### CODE REVIEW

**Comment by Tormund** [comment]
Any suggestions?

File: compiler/extccomp.nim [hunk 1]
```diff
@@ -766,14 +766,29 @@ proc callCCompiler*(projectfile: string) =
...
       add(objfiles, quoteShell(objFile))
 
     linkCmd = getLinkCmd(projectfile, objfiles)
     if optCompileOnly notin gGlobalOptions:
-      execLinkCmd(linkCmd)
+      when defined(windows):
+        let isCmdFileUsed = cCompiler == ccGcc and linkCmd.len >= 32_767
+        let cmdFile = projectfile & "_linker_cmd.txt"
+        if isCmdFileUsed:
+          let gccExeIdx = linkCmd.find("-o")
```
Comment (Araq) [inline compiler/extccomp.nim]: This is a hacky post processing step and the non-hacky solution is to change how ``execLinkCmd`` works.

**Comment by Tormund** [comment]
Workaround moved to execLinkCmd
And may also work for vcc ( but I can't test this now)
https://docs.microsoft.com/en-us/cpp/build/reference/at-specify-a-linker-response-file

**Comment by Araq** [comment]
Since we now have ``pathutils.relativeTo`` there is a much nicer way of fixing this.

**Pull request closed without merging**